docs/docs/reference/04-astro/index.md

```diff
@@ -31,52 +66,65 @@ export const authOpts: AstroAuthConfig = {
 
 export const { get, post } = AstroAuth(authOpts)
 ```
-
+Some OAuth Providers request a callback URL be submitted alongside requesting a Client ID, and Client Secret. 
 The callback URL used by the [providers](https://authjs.dev/reference/core/modules/providers) must be set to the following, unless you override the `prefix` field in `authOpts`:
 ```
 [origin]/api/auth/callback/[provider]
 ```
 
-When deploying your app outside Vercel, set the `AUTH_TRUST_HOST` variable to `true` for other hosting providers like Cloudflare Pages or Netlify.
-
 
-## Auth Secret
-
-Generate an auth secret by running `openssl rand -hex 32` or checking out [generate-secret.vercel.app](https://generate-secret.vercel.app/32), then set it as an environment variable:
-
-```sh
-AUTH_SECRET=<auth-secret>
-```
 
 ## Sign in & Sign out
 
-Astro Auth exposes two ways to sign in and out. The `signIn` and `signOut` methods can be imported dynamically in an inline script.
+Astro Auth exposes two ways to sign in and out. Inline scripts and Astro Components.
+
+### With Inline script tags
 
-```html
-<button id="login">Login</button>
-<button id="logout">Logout</button>
+The `signIn` and `signOut` methods can be imported dynamically in an inline script.
 
-<script>
-  let { signIn, signOut } = await import("@auth/astro/client")
-  document.querySelector("#login").onclick = () => signIn("github")
-  document.querySelector("#logout").onclick = () => signOut()
-</script>
+```astro
+---
+---
+<html>
+<body>
+  <button id="login">Login</button>
+  <button id="logout">Logout</button>
+
+  <script>
+    let { signIn, signOut } = await import("@auth/astro/client")
+    document.querySelector("#login").onclick = () => signIn("github")
+    document.querySelector("#logout").onclick = () => signOut()
+  </script>
+</body>
+</html>
 ```
+### With @auth/astro's Components
 
-Alternatively, you can use the `SignIn` and `SignOut` button components.
+Alternatively, you can use the `SignIn` and `SignOut` button components provided by `@auth/astro/components` importing them into your Astro [component's script](https://docs.astro.build/en/core-concepts/astro-components/#the-component-script) 
 
-```html
-<SignIn provider="github" />
-<SignOut />
+```astro
```

Comment:
These docs don't support astro syntax highlighting, so we use the next best option for each situation.

```suggestion
```jsx
```